src/api/rpc_api/state.rs

```diff
@@ -263,6 +284,35 @@ where
 		Ok(storage)
 	}
 
+	async fn get_all_storage_keys_paged_up_to_count(
+		&self,
+		storage_key_prefix: Option<StorageKey>,
+		count: u32,
+		start_key: Option<StorageKey>,
+		at_block: Option<Self::Hash>,
+	) -> Result<Vec<StorageKey>> {
+		let mut storage_keys: Vec<StorageKey> = Vec::new();
+		let mut still_todo = count;
+		let mut new_key = start_key;
+
+		while still_todo > 0 {
+			let new_count = if still_todo < STORAGE_KEYS_PAGED_MAX_COUNT {
+				still_todo
+			} else {
+				STORAGE_KEYS_PAGED_MAX_COUNT
+			};
+
+			let mut keys = self
```

Comment:
We need somewhere a "breakout" check. `get_storage_keys_paged` either returns up to the given `max_count` or all the keys with the given prefix, if there's less and `max_count`. So if there are fewer keys returned than the given count, we can break the loop - no more keys left to fetch.